doc/src/hexdoc_hexcasting/_templates/hexcasting_render.js.jinja

```diff
@@ -319,19 +319,80 @@ let options = {
 };
 
 function load_render(name) {
-  if (selected == name) {
+  /*if (selected == name) {
     return;
-  }
-
-  selected = name;
+  }*/
 
-  if (name == "animated") {
+  if (name == "animated" && selected != name) {
     load_animated();
   } else {
     render_images(options[name], last_palette, name + "-settings");
   }
+
+  selected = name;
+}
+
+function checkEqual(ob1, ob2) {
+  if (ob1 == ob2) {
+    return true;
+  } else if (ob1 == null || ob2 == null) {
+    return false;
+  } else if (Object.keys(ob1).length != Object.keys(ob2).length) {
+    return false;
+  }
+
+  for (let key in ob1) {
+    if (!(key in ob2)) {
+      return false;
+    } else if (typeof ob1[key] != typeof ob2[key]) {
+      return false;
+    } else if (typeof ob1[key] == "object") {
+      if (!checkEqual(ob1[key], ob2[key])) {
+        return false;
+      }
+    } else if (ob1[key] != ob2[key]) {
+      return false;
+    }
+  }
+  return true;
 }
 
+let autoUpdateInterval = null;
+function autoUpdateRenderer(interval) {
+  let real_interval = interval;
+  if (autoUpdateInterval != null) {
+    clearInterval(autoUpdateInterval);
+  }
+
+  if (real_interval == null) {
+    real_interval = 500;
+  } else if (real_interval <= 0) {
+    return;
+  }
```

Comment:
this feels confusing; why is real_interval separate from interval?

```js
interval ??= 500;
if (interval < 0) {
  console.log(`Invalid interval for autoUpdateRenderer: ${interval}`);
  return;
}
```

Reply:
I think I was in a Rust mindset and forgot that I could mutate the parameter. I replaced it with a default parameter instead